flink-runtime/src/test/java/org/apache/flink/runtime/asyncprocessing/EpochManagerTest.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.asyncprocessing;
+
+import org.apache.flink.runtime.asyncprocessing.EpochManager.Epoch;
+import org.apache.flink.runtime.asyncprocessing.EpochManager.EpochStatus;
+import org.apache.flink.runtime.asyncprocessing.EpochManager.ParallelMode;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+/** Test for Epoch Manager. */
+class EpochManagerTest {
+    @Test
+    void testBasic() {
+        EpochManager epochManager = new EpochManager(null);
+        Epoch epoch1 = epochManager.onRecord();
+        Epoch epoch2 = epochManager.onRecord();
+        assertThat(epoch1).isEqualTo(epoch2);
+        assertThat(epoch1.ongoingRecordCount).isEqualTo(2);
+        AtomicInteger output = new AtomicInteger(0);
+        epochManager.onNonRecord(
+                () -> {
+                    output.incrementAndGet();
+                },
+                ParallelMode.PARALLEL_BETWEEN_EPOCH);
```

Comment:
Despite that `SERIAL_BETWEEN_EPOCH` have been covered in `AsyncExecutionControllerTest`, it might still be better to make `EpochManagerTest` cover this situation as well.

Reply:
Constructing aec is troublesome and will bring a lot of duplicate code, so I put the test of `SERIAL_BETWEEN_EPOCH ` in `AsyncExecutionControllerTest `. I added a test for mix epoch mode in `AsyncExecutionControllerTest `.